tests/src/main/java/org/hibernate/beanvalidation/tck/util/ConstraintViolationAssert.java

```diff
@@ -232,12 +232,42 @@ public static PathExpectation pathWith() {
 		return new PathExpectation();
 	}
 
+	public static ViolationExpectation violationWith() {
+		return new ViolationExpectation();
+	}
+
 	public static class ConstraintViolationSetAssert extends IterableAssert<ConstraintViolation<?>> {
 
 		protected ConstraintViolationSetAssert(Set<? extends ConstraintViolation<?>> actualViolations) {
 			super( actualViolations );
 		}
 
+		public void containsOnlyViolations(ViolationExpectation... expectedViolations) {
+			isNotNull();
+
+			List<ViolationExpectation> actualViolations = new ArrayList<>();
+
+			ViolationExpectationPropertiesToTest referencePropertiesToTest;
+			if ( expectedViolations.length == 0 ) {
+				referencePropertiesToTest = ViolationExpectationPropertiesToTest.all();
+			}
+			else {
+				referencePropertiesToTest = expectedViolations[0].propertiesToTest;
+				for ( ViolationExpectation expectedViolation : expectedViolations ) {
+					if ( !referencePropertiesToTest.equals( expectedViolation.propertiesToTest ) ) {
+						throw new IllegalArgumentException( String.format( "Expected violations are not testing the properties consistently: %1$s - %2$s",
```

Comment:
Depending on what we declare in the expected violations, we will test some properties and not others. As the test is based on `containsExactlyInAnyOrder`, we use `hashCode` and `equals` to compare the violations, the result of these depending on the tested properties .

Thus I mandate all the expected violations to be consistent in what they are testing (i.e. you can't test just the constraint type in one ViolationExpectation and the message in another).